docs/specs/servers/blockchain/blockchain-permissions-api.md

```diff
@@ -174,3 +174,51 @@ The POST request body should be in JSON format and following schema:
 
 * `400 Bad request` - Wrong format in request.
 * `401 Unauthorized` - Wrong signature.
+
+## Co-signing
+
+Co-signing request
+
+`POST /v1/sessions/{address}/sign?projectId={projectId}`
+
+* `address` - CAIP-10 address format.
+* `projectId` - Required. The project identifier.
+
+### Request body:
+
+The POST request body should be in JSON format and following schema:
+
+```typescript
+{
+	pci: String, // Permission controller identifier.
+	userOp: // ERC-4337 pseudo transaction object:
+  {
+    sender: string, // The address of the smart contract account.
+    nonce: number, // Anti-replay protection; also used as the salt for first-time account creation.
+    initCode: string, // Code used to deploy the account if not yet on-chain.
+    callData: string, // Data that's passed to the sender for execution.
+    callGasLimit: number, // Gas limit for execution phase.
+    verificationGasLimit: number, // Gas limit for verification phase.
+    preVerificationGas: number, // Gas to compensate the bundler.
+    maxFeePerGas: number, // Maximum fee per gas (similar to EIP-1559(opens in a new tab) max_fee_per_gas).
+    maxPriorityFeePerGas: number, // Maximum priority fee per gas (similar to EIP-1559 max_priority_fee_per_gas).
+    paymasterAndData: string, // Paymaster Contract address and any extra data required for verification and execution (empty for self-sponsored transaction).
+    signature: string // Signature of all `userOp` fields hashed by keccak256 and signed by the key provided during the permission creation.
+  }
```

Comment:
it is expected from Dapp to build the userOp and then call this API to validate permissions and submit ? or co-signer will build the userOp and submit?

Reply:
Yes. The dapp needs to build userOp and sign it first.

Reply:
In that case bundler gas estimation, paymasterData and all is also done by the dapp,
Dapp will sign and set the signature on userOp.signature field and forward the userOp to co-signer 
co-signer will sign the userOp and submit it.
Q. Will co-signer use the userOpBuilder to get the final signature? If yes how will it pass the 2 signatures it have to `getSignatureWithContext`?